app/screens/HomePage.js

```diff
@@ -89,13 +97,41 @@ export default class HomePage extends Component<{}> {
         Database.getMyExperimentInfo(this.state.my_experiment_data.experiment_id).then((data) => {
             this.setState({
                 experiment_info: data
-            });
+            }, this.hasUserCompletedExperiment);
         }).catch((error) => {
             console.log("Error because: "+error);
         });
     }
 
 
+    /**
+     * Check if user has completed experiment, if yes, redirect.
+     */
+    hasUserCompletedExperiment() {
+        console.log('aquiiiii');
+        data = [];
+        if(this.state.my_experiment_data && this.state.my_experiment_data.reactions) {
+            Object.keys(this.state.my_experiment_data.reactions).forEach((key, index) => {
+
+                var info = (this.state.my_experiment_data.reactions[key]);
+                var val = {x: key, y: info.reaction};
+                data.push(val);
+            });
+        }
+
+        //Update to 14
+        if(data.length >= 14){
```

Comment:
was comparing by date too difficult? I think Date.parse would be much cleaner here. 

Then, to find the difference, just do:

var date1 = new Date("7/13/2010");
var date2 = new Date("12/15/2010");
var timeDiff = Math.abs(date2.getTime() - date1.getTime());
var diffDays = Math.ceil(timeDiff / (1000 * 3600 * 24)); 

Would like to see this change in the pr before approval.

Reply:
resolved.